assets/tests/TaskSchemaValidationService.js

```diff
@@ -38,11 +38,13 @@ tie.factory('TaskSchemaValidationService', [
         var instructions = task.getInstructions();
         return instructions.length > 0;
       },
-      verifyInstructionsIsArrayOfStrings: function(task) {
+      verifyInstructionsIsAssociativeArray: function(task) {
```

Comment:
You should also verify that the types are within the types that you allow, not just type: "HAADASD" or whatever.